src/UI/Root/Components/Sidebar/Navigation/NavigationItem.tsx

```diff
@@ -68,6 +76,42 @@ const ExternalItem: React.FC<Label & Url> = ({ label, url }) => (
   </NavItem>
 );
 
+const CompileReportItem: React.FC<Label & Url> = ({ label, url }) => {
+  const { queryResolver } = useContext(DependencyContext);
+  const [data, retry] = queryResolver.useContinuous<"GetCompilationState">({
+    kind: "GetCompilationState",
+  });
+
+  useEffect(() => {
```

Comment:
Am I understanding it right, that you are adding and removing the listener on every API/query call?

Reply:
that return is useEffect version of componentsDidUnmount